File: packages/dart/lib/src/word/rand_sound.dart
import 'dart:math';

import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// Sounds and voices, and the words a language has for them.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.sound] to `randWordDetails` — Dart has
/// no overloads, and a second function per theme would be a second wall of them.
///
/// ```dart
/// randSound(language: WordLanguage.ko, count: 3); // [속삭임, 함성, 바스락]
/// randSound(language: WordLanguage.en, count: 3); // [Whisper, Chime, Rustle]
/// ```
List<String> randSound({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  RandVocabulary vocabulary = RandVocabulary.full,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,

  /// Where the randomness comes from: `Random.secure()` for a value nobody may
  /// predict, `Random(42)` for one that has to come out the same every run.
  Random? random,
}) => randWord(
  language: language,
  theme: WordTheme.sound,
  count: count,
  realism: realism,
  vocabulary: vocabulary,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
  random: random,
);

File: packages/javascript/lib/word/randSound.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * Sounds and voices, and the words a language has for them.
 *
 * `randWord({ theme: 'sound' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randSound({ language: 'ko', count: 3 }); // ['속삭임', '함성', '바스락']
 * randSound({ language: 'en', count: 3 }); // ['Whisper', 'Chime', 'Rustle']
 */
export function randSound(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Sound words with the language and theme each one came from.
 *
 * @example
 * randSound({ language: 'ko', output: 'detail' });
 * // [{ word: '속삭임', language: 'ko', theme: 'sound' }]
 */
export function randSound(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randSound(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('sound', options);
}
